### PR TITLE
Add nodeSelector provisioner;

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Adds provisioner which can apply optionally configured node selector to workspace pod.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14562

#### Docs PR

https://github.com/eclipse/che-docs/pull/1479

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

**Review by metlos — APPROVED** [review]

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class NodeSelectorProvisioner implements ConfigurationProvisioner {
+
+  private final Map<String, String> nodeSelectorAttributes;
+
+  @Inject
+  public NodeSelectorProvisioner(
+      @Nullable @Named("che.workspace.pod.node_selector") String nodeSelectorProperty) {
+    this.nodeSelectorAttributes =
+        nodeSelectorProperty != null
+            ? Splitter.on(",").withKeyValueSeparator("=").split(nodeSelectorProperty)
```
Comment (amisevsk) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java:35]: Can this throw `IllegalArgumentException`, if e.g. the admin misses a comma in the property string?

**Comment by mshaposhnik** [comment]
I have tried that on real multi-node multi -zone cluster, and get the following error:
```
Error: Failed to run the workspace: "Plugins installation process failed. Error: Unrecoverable event occurred: 'FailedScheduling', '0/6 nodes are available: 1 node(s) had volume node affinity conflict, 5 node(s) didn't match node selector.', 'workspacevuoho6h1f7egvki9.che-plugin-broker'"
```

Looks like some problems with PV/PVC on broker...

**Comment by mshaposhnik** [comment]
[ci-build]

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java (hunk 1 above)
Comment (skabashnyuk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java:35]: I didn't found such functionality in Splitter. Does it worth rewriting the whole logic to do that?
or we can add condition below if `nodeSelectorProperty.size >0 and nodeSelectorAttributes.size == 0` then throw IllegalArgumentException("invalid format of nodeSelectorProperty") .
wdyt?

Comment (skabashnyuk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java:35]: I was wrong
```
MapSplitter.
     * @throws IllegalArgumentException if the specified sequence does not split into valid map
     *     entries, or if there are duplicate keys
```

**Comment by mshaposhnik** [comment]
[ci-build]

**Review by sparkoo — APPROVED** [review]

**Review by amisevsk — APPROVED** [review]
I'm OK with merging as-is, but want to make sure we're on the same page with the comment below -- I know we crashloop when other properties are misconfigured, so it's probably fine.

**Pull request merged**